main.go

```diff
@@ -58,29 +53,37 @@ func init() {
 }
 
 func main() {
-	var metricsAddr string
-	var enableLeaderElection bool
-	var probeAddr string
-	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
-	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
-	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
-		"Enable leader election for controller manager. "+
-			"Enabling this will ensure there is only one active controller manager.")
-	opts := zap.Options{
-		Development: true,
+	err := conf.Load()
+	if err != nil {
+		setupLog.Error(err, "failed to load the configuration")
+		os.Exit(1)
+	}
+
+	if err := conf.Get().Validate(); err != nil {
+		setupLog.Error(err, "invalid configuration")
+		os.Exit(1)
 	}
-	opts.BindFlags(flag.CommandLine)
-	flag.Parse()
 
+	var level zapcore.Level
+	err = level.UnmarshalText([]byte(conf.Get().LogLevel))
+	if err != nil {
+		setupLog.Error(err, "unable to unmarshal log level", "log level", conf.Get().LogLevel)
+		os.Exit(1)
+	}
+	opts := zap.Options{}
+	opts.Level = level
 	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
 
+	setupLog = ctrl.Log
+	setupLog.Info("Started with configuration", "configuration", conf.Get())
+
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
 		Scheme:                 scheme,
-		MetricsBindAddress:     metricsAddr,
-		Port:                   9443,
-		HealthProbeBindAddress: probeAddr,
-		LeaderElection:         enableLeaderElection,
-		LeaderElectionID:       "bfdcaedc.osbuilder.project-flotta.io",
+		MetricsBindAddress:     conf.Get().MetricsAddr,
```

Comment:
Please create a `configuration:=conf.Get()` and use it throughout the method. I think it would be more readable.

Reply:
Once I externalized the `GlobalConf` this becomes less relevant, right?

Reply:
Yeah, it's good now.